### ISSUE TITLE
train_grpo.py 代码分支有个 bug

### ISSUE THREAD

**Issue by somday-ican** [jingyaogong/minimind#750]
这版 MiniMind 的 train_grpo.py 代码分支有个 bug。

我对照源码看了下，问题点在这里：

当前仓库的 train_grpo.py 在 torch rollout 且 use_moe=0 时，会走这个分支：
per_token_logps = rollout_result.per_token_logps
这个 per_token_logps 很可能是 rollout 阶段算出来的、不带梯度 的张量
后面 loss.backward() 时，就会报这个错：
element 0 of tensors does not require grad

**Comment by jingyaogong** [comment]
已经更新修复，拉取最新代码
